state/models.go

```diff
@@ -734,3 +737,179 @@ func (w *Record) Equal(other Record) bool {
 func (w *Record) String() string {
 	return fmt.Sprintf("%s-%s", w.EventSource, w.EventName)
 }
+
+const TemplatePayloadKey = "template_payload"
+
+type TemplatePayload map[string]interface{}
+
+type TemplateExecutionRequest struct {
+	TemplatePayload TemplatePayload `json:"template_payload"`
+	ExecutionRequestCommon
+}
+
+func (t TemplateExecutionRequest) GetExecutionRequestCommon() *ExecutionRequestCommon {
+	return &t.ExecutionRequestCommon
+}
+
+func (t TemplateExecutionRequest) GetExecutionRequestCustom() *map[string]interface{} {
+	return &map[string]interface{}{
+		TemplatePayloadKey: t.TemplatePayload,
+	}
+}
+
+type Template struct {
+	TemplateID      string `json:"template_id"`
+	Type            string `json:"type"`
```

Comment:
maybe we should use `template_name` - similar to `template_id` `type` tends to be a keyword in few different languages.

Reply:
👍